src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java

```diff
@@ -447,6 +504,11 @@ public void postUpdate( final List<IAEItemStack> list, final byte ref )
 		this.setScrollBar();
 	}
 
+	int getMaxRows()
+	{
+		return AEConfig.instance().getConfigManager().getSetting( Settings.TERMINAL_STYLE ) == TerminalStyle.SMALL ? 5 : Integer.MAX_VALUE;
```

Comment:
Should handle also the sanitation part. But for performance maybe keep `this.rows` and rename this to something like `calculateRows()` and call it during init?